README.md

```diff
@@ -95,37 +148,36 @@ public class StringExample {
 
         // Instantiate the SDK
         final AwsCrypto crypto = AwsCrypto.standard();
-
-        // Set up the master key provider
-        final KmsMasterKeyProvider prov = KmsMasterKeyProvider.builder().buildStrict(keyArn);
-
+        
+        // Create the AWS KMS keyring.
+        // We create a multi keyring, as this interface creates the KMS client for us automatically.
+        final MaterialProviders materialProviders = MaterialProviders.builder()
+                .MaterialProvidersConfig(MaterialProvidersConfig.builder().build())
+                .build();
+        final CreateAwsKmsMultiKeyringInput keyringInput = 
+                CreateAwsKmsMultiKeyringInput.builder().generator(keyArn).build();
+        final IKeyring kmsKeyring = materialProviders.CreateAwsKmsMultiKeyring(keyringInput);
+        
         // Set up the encryption context
         // NOTE: Encrypted data should have associated encryption context
         // to protect its integrity. This example uses placeholder values.
         // For more information about the encryption context, see
         // https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/concepts.html#encryption-context
-        final Map<String, String> context = Collections.singletonMap("ExampleContextKey", "ExampleContextValue");
+        final Map<String, String> encryptionContext = Collections.singletonMap("ExampleContextKey", "ExampleContextValue");
 
         // Encrypt the data
-        //        
-        final CryptoResult<byte[], KmsMasterKey> encryptResult = crypto.encryptData(prov, plaintext.getBytes(StandardCharsets.UTF_8), context);
+        final CryptoResult<byte[], ?> encryptResult = crypto.encryptData(kmsKeyring, plaintext.getBytes(StandardCharsets.UTF_8), encryptionContext);
```

Comment:
The `?` is a wild card to keep `encryptData` API compatible with previous releases.

ESDK-Java's CryptoResult includes a definition of the  actual Cryptographic Material Provider (MasterKey or MasterKeyProvider) as a Generic that can be refined at class initialization.

I am not sure if we ever refactored that `CryptoResult` class so
 `IKeyring` or `ICryptographicMaterialProvider` could refine the Generic...

Reply:
We clearly did not:
https://github.com/aws/aws-encryption-sdk-java/blob/master/src/main/java/com/amazonaws/encryptionsdk/CryptoResult.java#L31

This means that following APIs are now broken for Keyring users:
- [`CryptoResult.getMasterKeys`](https://github.com/aws/aws-encryption-sdk-java/blob/master/src/main/java/com/amazonaws/encryptionsdk/CryptoResult.java#L67-L69)
- [`CryptoResult.getMasterKeyIds`](https://github.com/aws/aws-encryption-sdk-java/blob/master/src/main/java/com/amazonaws/encryptionsdk/CryptoResult.java#L71-L79)

Which... is not Great. 
@imabhichow we should bring this up at Stand up. 

Those APIs are NOT critical; one of them is even marked as deprecated.

Reply:
> Which... is not Great.

Looking into the details here, this is not bad or even an issue.

`K` is would be an explicit MasterKey.
If no MaterKey or MasterKeyProviders are used, 
then no one SHOULD expect a list of MasterKeyIDs to be populated.

We should probably write a test to assert that nothing wild happens,
like an Exception.

Reply:
When using keyrings, the `CryptoResult.getMasterKeyIds` method returns an empty list during encryption.

However, during decryption, [CryptoResult.getMasterKeyIds](https://github.com/aws/aws-encryption-sdk-java/blob/master/src/main/java/com/amazonaws/encryptionsdk/CryptoResult.java#L71-L79) throws a NULL POINTER EXCEPTION. This could be due to returning a [DataKey](https://github.com/aws/aws-encryption-sdk-java/blob/905385d790011e2402543c6eddef400350b1229d/src/main/java/com/amazonaws/encryptionsdk/model/DecryptionMaterialsHandler.java#L44) with the master key set to null. I suppose we should expect similar behavior to encryption which is returning an empty llist of masterkeys. I'll try to look more into this.

Reply:
We should have, yes.
And we should cut a PR to prevent the NPE.

Reply:
With this [pull request](https://github.com/aws/aws-encryption-sdk-java/pull/1976), [CryptoResult.getMasterKeyIds](https://github.com/aws/aws-encryption-sdk-java/blob/master/src/main/java/com/amazonaws/encryptionsdk/CryptoResult.java#L71-L79) is now deprecated and will return an empty list when we use keyrings.